### PR TITLE
[GLUTEN-8074][CH] Fix adjust output constant column

### PR DESCRIPTION
**Pull request by lwz9103**
## What changes were proposed in this pull request?

(Please fill in changes proposed in this fix)

(Fixes: \#8074)

### CODE REVIEW

File: cpp-ch/local-engine/Parser/SerializedPlanParser.cpp [hunk 1]
```diff
@@ -160,20 +160,31 @@ void SerializedPlanParser::adjustOutput(const DB::QueryPlanPtr & query_plan, con
...
+                {
+                    const DB::ContextPtr context = DB::CurrentThread::get().getQueryContext();
+                    const FunctionOverloadResolverPtr & cast_resolver = FunctionFactory::instance().get("CAST", context);
+                    const DataTypePtr string_type = DB::DataTypeFactory::instance().get("String");
+                    ColumnWithTypeAndName to_type_column = {string_type->createColumnConst(1, final_type->getName()), string_type, "__cast_const__"};
+                    FunctionBasePtr cast_function = cast_resolver->build({origin_column, to_type_column});
+                    ColumnPtr cast_col = cast_function->execute({origin_column, to_type_column}, final_type, 1);
+                    Field cast_field;
+                    cast_col->get(0, cast_field);
+                    ColumnWithTypeAndName final_column(final_type->createColumnConst(0, cast_field), final_type, origin_column.name);
```
Comment (taiyang-li) [inline cpp-ch/local-engine/Parser/SerializedPlanParser.cpp:174]: maybe we can wrap `ColumnConst` around `cast_col` directly without extracting field from it.

File: cpp-ch/local-engine/Parser/SerializedPlanParser.cpp [hunk 2]
```diff
@@ -160,20 +160,31 @@ void SerializedPlanParser::adjustOutput(const DB::QueryPlanPtr & query_plan, con
...
-                    need_const ? final_type->createColumnConst(0, assert_cast<const ColumnConst &>(*origin_column.column).getField())
-                               : final_type->createColumn(),
-                    final_type,
-                    origin_column.name);
-                final_columns.emplace_back(std::move(final_column));
+                if (need_const)
+                {
+                    const DB::ContextPtr context = DB::CurrentThread::get().getQueryContext();
+                    const FunctionOverloadResolverPtr & cast_resolver = FunctionFactory::instance().get("CAST", context);
+                    const DataTypePtr string_type = DB::DataTypeFactory::instance().get("String");
```
Comment (taiyang-li) [inline cpp-ch/local-engine/Parser/SerializedPlanParser.cpp:168]: DataTypePtr string_type = std::make_shared<DataTypeString>();

File: cpp-ch/local-engine/Parser/SerializedPlanParser.cpp (hunk 1 above)
Comment (lwz9103) [reply cpp-ch/local-engine/Parser/SerializedPlanParser.cpp:172]: thanks, it's more concise and clear

File: cpp-ch/local-engine/Parser/SerializedPlanParser.cpp (hunk 2 above)
Comment (lwz9103) [reply cpp-ch/local-engine/Parser/SerializedPlanParser.cpp]: thanks, it's more concise and clear

**Review by taiyang-li — APPROVED** [review]
LGTM

**Review by zzcclp — APPROVED** [review]

**Pull request merged**